cmd/api/src/database/sso_providers_test.go

```diff
@@ -43,6 +44,30 @@ func TestBloodhoundDB_CreateSSOProvider(t *testing.T) {
 		assert.Equal(t, "Bloodhound Gang", result.Name)
 		assert.Equal(t, "bloodhound-gang", result.Slug)
 		assert.Equal(t, model.SessionAuthProviderSAML, result.Type)
+
+		provider, err := dbInst.GetSSOProvider(testCtx, int(result.ID))
+		require.NoError(t, err)
+		assert.Equal(t, model.SessionAuthProviderSAML, provider.Type)
+	})
+}
+
+func TestBloodhoundDB_DeleteSSOProvider(t *testing.T) {
+	var (
+		testCtx = context.Background()
+		dbInst  = integration.SetupDB(t)
+	)
+	defer dbInst.Close(testCtx)
+
+	t.Run("successfully delete an SSO provider", func(t *testing.T) {
```

Comment:
Would it be possible to add a user to the saml provider and ensure that it properly sets it to null on both fields, ssoProviderId and samlProviderId?

And same with an OIDC provider / OIDC user?